src/librustc/ty/fold.rs

```diff
@@ -184,8 +184,10 @@ pub trait TypeVisitor<'tcx> : Sized {
         t.super_visit_with(self)
     }
 
+    // Can be overridden, then call `self.super_xxx` to revert back to the
+    // default behavior.
     fn visit_ty(&mut self, t: Ty<'tcx>) -> bool {
-        t.super_visit_with(self)
+        self.super_ty(t)
```

Comment:
oh, my bad, I should have written `super_visit_with` I suppose. Let's revert this part of the diff and just have the code below invoke `super_visit_with` — although I do prefer (these days) to make matching `super_ty` methods and so forth, it doesn't seem to be the standard in `TypeVisitor`.